pinot-tools/src/main/java/org/apache/pinot/tools/realtime/provisioning/MemoryEstimator.java

```diff
@@ -107,6 +129,15 @@ public MemoryEstimator(TableConfig tableConfig, File sampleCompletedSegment, int
     _tableDataDir.mkdir();
   }
 
+  public MemoryEstimator(TableConfig tableConfig, File dataCharacteristicsFile, File schemaFile, int ingestionRate,
+      long maxUsableHostMemory, int tableRetentionHours) {
+    this(tableConfig,
+        generateCompletedSegment(dataCharacteristicsFile, schemaFile, tableConfig),
```

Comment:
Can we add a small test (or ensure manually) that the file location of the generated completed segment returned by generateCompletedSegment() is in the same segment file hierarchy as taken by the other constructor

Reply:
Why do we want to do that? We generate it in `java.io.tmpdir` so it gets deleted automatically.

Reply:
Just wanted to ensure that the new code (added in this PR) to generated the completed segment generates it in the correct directory hierarchy as expected by the second constructor. Looks like your unit test already covers it.